File: two_pointers/two_sum_ii.py
# Q:167 - https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1

        while left < right:
            sum = numbers[left] + numbers[right]

            if sum < target:
                left += 1
            elif sum > target:
                right -= 1
            else:
                return [left + 1, right + 1]


c = Solution()

numbers1 = [2,7,11,15]
target1 = 9

numbers2 = [2,3,4]
target2 = 6

numbers3 = [-1,0]
target3 = -1

print(c.twoSum(numbers1, target1))
print(c.twoSum(numbers2, target2))
print(c.twoSum(numbers3, target3))

File: two_pointers/two_sum_ii.js
// Q:167 - https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let left = 0, right = numbers.length - 1

    while (left < right) {
        const sum = numbers[left] + numbers[right];
        
        if (sum > target) {
            right -= 1;
        } else if (sum < target) {
            left += 1
        } else {
            return [left + 1, right + 1]
        }
    }

};